EIPS/eip-7425.md

```diff
@@ -0,0 +1,175 @@
+---
+eip: 7425
+title: Tokenized Reserve
+description: Reserve with user auditing
+author: Jimmy Debe (@jimstir)
+discussions-to: https://ethereum-magicians.org/t/eip-7425-tokenized-reserve/15297
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-30
+requires: 20, 4626
+---
+
+## Abstract
+
+A proposal for a tokenized reserve mechanism. The reserve allows an audit of on-chain actions of the owner. Using [ERC-4626](../EIPS/eip-4626.md), stakeholders can create shares to show support for actions in the reserve.
+
+## Motivation
+
+Tokenized reserves are an extension of tokenized vaults. The goal is to create a reserve similar to a real world reserve an entity has as a backup incase regular funds run low. In the real world, an entity will have certain criteria to access reserve funds. In a decentralized environment, an entity can incorporate stakeholders into their criteria. Hopefully this will help entities who participate in decentralized environments to be transparent with stakeholders.
+
+## Specification
+
+### Definitions:
+
+	- owner: The creator of the reserve.
+	- user: Stakeholders of specific proposals
+	- reserve: The tokenized reserve contract
+	- proposal: Occurs when the owner wants a withdrawal from contract
+ 
+### Constructor:
+ 
+ 	- name: ERC-20 token name
+  	- ticker: ERC-20 ticker
+   	- asset: ERC-4626 underlying ERC-20 address
+	- rAuth: Primary authorized user
+	- rOwner: Owner of the Reserve
+ 
+### Interface
+    
+``` solidity
+    interface TokenReserve{
+	    // @dev proposals event
+	    event proposals(
+	    	address indexed token,
+	    	uint256 indexed proposalNum,
+	    	uint256 indexed amount,
+	    	address recipient
+	    );
+	    /// @dev owner deposit event
+	    event depositR(
+	    	address indexed token,
+	    	uint256 indexed amount,
+	    	uint256 indexed time,
+	    	uint256 count
+	    );
+	    /** @dev Owner accounting struct
+	    * @param time Deposit time
+	    * @param token Address of ERC-20 token
+	    * @param deposit Amount of owner deposit
+	    */
+	    struct ownerAccount{
+	    	uint256 time;
+	    	address token;
+	    	uint256 deposit;
+	    }
+	    /** @dev User accounting struct
+	    * @param proposal Specific participating proposal number
+	    * @param deposit Amount user deposited
+	    * @param withdrew Amount of user withdrew
+	    */
+	    struct userAccount{
+	    	uint256 proposal;
+	    	uint256 deposit;
+	    	uint256 withdrew;
+	    }
```

Comment:
This struct doesn't seem used in the public members of this interface. EIPs should only standardize the externally visible properties of a contract, and leave the implementation details up to the implementer.

Reply:
I removed the structs and added getter functions. As you mentioned, I leave it up to the implementor to choose whether to use structs or variables for the getter functions.